### ISSUE TITLE
Errors on resource deletion when provider alias used

### ISSUE THREAD

**Issue by ImperialXT** [terraform-providers/terraform-provider-datadog#342]
When a resource is destroyed that has a provider declared for it e.g `datadog.test` errors are thrown after the resrouce has been destroyed about missing required arguments. Namely api_key and app_key. This behaviour is not seen on resources that do not have a provider declared.

```
datadog_monitor.test: Destroying... [id=12235480]
datadog_monitor.test: Destruction complete after 1s

Error: Missing required argument

The argument "api_key" is required, but was not set.


Error: Missing required argument

The argument "app_key" is required, but was not set.
```
### Terraform Version
- v0.12.12
### Provider Version
- v2.4.0
### Affected Resource(s)
- datadog_monitor

### Terraform Configuration Files
```hcl
provider "datadog" {
	alias = "test"
	version = "~> 2.4"
	api_key = "<api_key>"
	app_key = "<app_key>"
}

resource "datadog_monitor" "test" {
	provider = datadog.test
	name = "example for issue in github"
	type = "metric alert"
	query = "avg(last_1h):avg:aws.ec2.cpu{environment:foo,host:foo} by {host} > 4"
	message = "this is a test"
}
```

### Debug Output
https://gist.github.com/ImperialXT/7f3d32e64936ded03a3c254448ee95fd
### Expected Behavior
No errors thrown.

### Actual Behavior

The monitor was destroyed but errors were thrown.
```
An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  - destroy

Terraform will perform the following actions:

  # datadog_monitor.test will be destroyed
  - resource "datadog_monitor" "test" {
      - evaluation_delay    = 0 -> null
      - id                  = "12235480" -> null
      - include_tags        = true -> null
      - locked              = false -> null
      - message             = "this is a test" -> null
      - name                = "example for issue in github" -> null
      - new_host_delay      = 300 -> null
      - no_data_timeframe   = 10 -> null
      - notify_audit        = false -> null
      - notify_no_data      = false -> null
      - query               = "avg(last_1h):avg:aws.ec2.cpu{environment:foo,host:foo} by {host} > 4" -> null
      - renotify_interval   = 0 -> null
      - require_full_window = true -> null
      - silenced            = {} -> null
      - tags                = [] -> null
      - threshold_windows   = {} -> null
      - thresholds          = {} -> null
      - timeout_h           = 0 -> null
      - type                = "metric alert" -> null
    }

Plan: 0 to add, 0 to change, 1 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

datadog_monitor.test: Destroying... [id=12235480]
datadog_monitor.test: Destruction complete after 1s

Error: Missing required argument

The argument "app_key" is required, but was not set.


Error: Missing required argument

The argument "api_key" is required, but was not set.
```

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`
2. `Comment out/remove the resource from .tf file`
3. `terraform apply`

### Important Factoids
I run multiple datadog providers to configure monitors in multiple accounts from the one repository.

**Comment by dabcoder** [comment]
Also reproducible on version `2.5.0` of the provider. This issue does not occur if keys are set in env vars and not in the `tf` file. Still looking into this.

**Comment by roiavidan** [comment]
Still reproducible on the latest version (v2.7.0). After destroying all other resources, getting an error:

```
Error: Invalid or missing credentials provided to the Datadog Provider. Please confirm your API and APP keys are valid and see https://terraform.io/docs/providers/datadog/index.html for more information on providing credentials for the Datadog Provider

  on .terraform/modules/backend/monitors.tf line 1, in provider "datadog":
   1: provider "datadog" {
```

Terraform Version: 0.12.20
Provider Version: 2.7.0

I have tried removing the `api_key` and `app_key` arguments from the provider definition and putting the values into the environment, however, I then get a schema error saying those arguments are missing.

**Comment by nmuesch** [comment]
Hey @ImperialXT and @roiavidan I apologize for the delay here. I took a look into this with the provided reproduction steps (thanks for that) and was able to reproduce on terraform version 0.12.12

After digging in, it looks like it may be the same issue as https://github.com/hashicorp/terraform/issues/23567
To test, I switched to a newer version of terraform (0.12.24) and this issue seems to be resolved on version v2.5.0 of the Datadog Provider.

Given that this appears to be a core issue that was resolved with newer terraform versions, I'll go ahead and close this out. Please let us know if you continue to experience any issues here.
